### PR TITLE
Start client on demand under supervisor

### PR DESCRIPTION
**Pull request by id**
One can use brod_client:start_link if they know what they are doing.

### CODE REVIEW

File: src/brod_sup.erl [hunk 1]
```diff
@@ -62,29 +62,37 @@
...
-%%        , [ { endpoints, [{"localhost", 9092}]}
-%%          , { config
-%%            , [ {restart_delay_seconds, 10}
-%%                %% @see brod:start_link_client/5 for more client configs
-%%              ]
-%%            }
+%%        , [ {endpoints, [{"localhost", 9092}]}
+%%          , {restart_delay_seconds, 10}
+%%          , {max_metadata_sock_retry, 1}
+%%          , {get_metadata_timout_seconds, 5}
```
Comment (zmstone) [inline src/brod_sup.erl]: timeout

File: src/brod_sup.erl [hunk 2]
```diff
@@ -62,29 +62,37 @@
...
+%%          , {reconnect_cool_down_seconds, 1}
+%%          , {allow_topic_auto_creation, true}
```
Comment (zmstone) [inline src/brod_sup.erl:94]: false

File: src/brod_sup.erl [hunk 3]
```diff
@@ -113,13 +132,19 @@ post_init(_) ->
...
+client_spec([], ClientId,  _Config) ->
+  Error = lists:flatten(
+            io_lib:format("No endpoints found in brod client '~p' config",
+                          [ClientId])),
+  exit(Error);
+client_spec(Endpoints, ClientId,  Config) ->
+  DelaySecs = proplists:get_value(restart_delay_seconds, Config,
                                   ?DEFAULT_CLIENT_RESTART_DELAY),
-  Config    = proplists:delete(restart_delay_seconds, Config0),
+  Config    = proplists:delete(restart_delay_seconds, Config),
```
Comment (zmstone) [inline src/brod_sup.erl]: it's proplists:delete here

File: src/brod_sup.erl (hunk 2 above)
Comment (id) [inline src/brod_sup.erl:94]: https://github.com/klarna/brod/blob/2.1-dev/src/brod_client.erl#L528

File: src/brod_sup.erl (hunk 3 above)
Comment (id) [inline src/brod_sup.erl]: /o

**Comment by zmstone** [comment]
squash the commits ?

**Pull request merged**